### PR TITLE
[WIP] Create part-1-the-engine.md

### PR DESCRIPTION
**Pull request by thangngoc89**

### CODE REVIEW

File: content/sketch/part-1-the-engine.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    -linkpkg -o my_app.byte my_app.ml
+```
+
+Next you call js_of_ocaml on the output bytecode like this:
+
+```
+js_of_ocaml my_app.byte
+```
+
+And you'll a `my_app.js` file ready to be embed in your website.
```
Comment (matthiaskern) [inline content/sketch/part-1-the-engine.md:48]: receive/get

File: content/sketch/part-1-the-engine.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+# Building instructions
+
+The code example in the post is [here](https://github.com/thangngoc89/blog/tree/master/content/sketch/examples).
```
Comment (matthiaskern) [inline content/sketch/part-1-the-engine.md:52]: I would mention what this example is for, you're straight away jumping to `Building instructions` without any former mention of it.

File: content/sketch/part-1-the-engine.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+```reason
+JsooTop.initialize();
+let () = Js.export_all(
+  [%js {
+    val execute = execute
+  }]
+);
+```
+
+The first line is for initialization of the toplevel, it does several things like polulating the toplevel environment, setting up look up paths,... but we don't need to worry about it right now.
```
Comment (matthiaskern) [inline content/sketch/part-1-the-engine.md:129]: populating, look-up

File: content/sketch/part-1-the-engine.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+Hello world from OCaml
+'- : unit = ()\n'
+
+> execute(`syntax error`);
+File "", line 1, characters 12-12:
+Error: Syntax error
+''
+```
+
+- It executes all code at once. We want it to return the result for each individual statement (they are called phrases in OCaml's term).
```
Comment (matthiaskern) [inline content/sketch/part-1-the-engine.md:165]: term->terminology

File: content/sketch/part-1-the-engine.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+> execute(`syntax error`);
+{ result: '',
+  stderr: 'File "", line 1, characters 12-12:\nError: Syntax error\n',
+  stdout: '' }
+```
+
+That looks more promising! We get `stderr` and `stdout` as return values. In Sketch UI, This information will be display inline like this
+
+![Sketch inline errors and values](./images/display-inline.png)
+
```
Comment (matthiaskern) [inline content/sketch/part-1-the-engine.md:243]: What about a final recap or something like "in the next post, you will learn about ..."?

**Comment by thangngoc89** [comment]
Merged locally in https://github.com/thangngoc89/blog/commit/26683c0d7035c5b98e9b15465e2aec482b4e9f27

**Pull request closed without merging**